Macaulay2/M2
# MonodromySolver v1.13

Summary of changes to the package MonodromySolver

  1. code refactoring and cleanup
    1. coreMonodromy: exit main loop if no edges are available (ie, there exist solutions with unknown correspondences)
    2. new default Edge selection strategy us to take first available edge.
    3. uniform, internal list of MonodromyOptions shared by methods that "do monodromy"
    4. internal function setStoppingCriterion added. stopping criteria may now depend on the homotopy graph
    5. better error handling in staticMonodromySolve
    6. better adherence to Macaulay2 style guide (compare dynamicMonodromySolve, old and new)
  2. complete rewrite of solveFamily, w/ modified documentation
  3. reorganized tests
    1. added tests for createSeedPair, Equivalencer, quadratic segment homotopy
    2. new "additional-tests" (not required for installation)
  4. remove uninvited warning messages, which used to be printed even when Verbose => false
  5. check linearity in parameters before auto-seeding, throw an error otherwise
  6. refactor trackEdge
    1. removed bug that assumed systems were invariant under scaling parameters by using quadratic segment homotopy

Other changes

1. Implementation of (jacobian, GateSystem) in extraNAGtypes package
